src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -29,9 +29,20 @@ public static (AddRangeResult result, bool moreChildrenToRight, List<PathWithAcc
             IReadOnlyList<byte[]> proofs = null
         )
         {
-            // TODO: Check the accounts boundaries and sorting
             if (accounts.Count == 0)
                 throw new ArgumentException("Cannot be empty.", nameof(accounts));
+            // Validate sorting and boundary range
+            for (int i = 1; i < accounts.Count; i++)
```

Comment:
Would be good to check if this is actually needed and what happens if the accounts are not sorted. While it is important to have sorted boundary nodes (as it is used for stitching and evaluating if we need more requests), setting the data on the trie doesn't seem to require strict order. Could you add tests?